Source/pfile.cpp

```diff
@@ -15,6 +18,43 @@ DEVILUTION_BEGIN_NAMESPACE
 #define PASSWORD_SINGLE "xrgyrkj1"
 #define PASSWORD_MULTI "szqnlsk1"
 
+namespace {
+
+std::string GetSavePath(DWORD save_num)
+{
+	std::string path = GetPrefPath();
+	if (gbIsSpawn) {
+		if (gbMaxPlayers <= 1) {
+			path.append("spawn");
+		} else {
+			path.append("share_");
+		}
+	} else {
+		if (gbMaxPlayers <= 1) {
+			path.append("single_");
+		} else {
+#ifdef HELLFIRE
+			path.append("hrinfo_");
+#else
+			path.append("multi_");
+#endif
+		}
+	}
+
+	char save_num_str[21];
+	snprintf(save_num_str, sizeof(save_num_str) / sizeof(char), "%d", save_num);
+	path.append(save_num_str);
+
+#ifdef HELLFIRE
+		path.append(".hsv");

```

Comment:
hellfire multiplayer files are .drv, unless we want to change that. im still missing a test file though

Reply:
Fixed